mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClustersHostManagerActor.java

```diff
@@ -328,10 +330,37 @@ private void onScaleResourceClusterRequest(ScaleResourceRequest req) {
     private void onUpgradeClusterContainersRequest(UpgradeClusterContainersRequest req) {
         log.info("Entering onScaleResourceClusterRequest: " + req);
         // [Notes] for scaling-up the request can go straight into provider to increase desire size.
-        // FOr scaling-down the decision requires getting idle hosts first.
-
-        pipe(this.resourceClusterProvider.upgradeContainerResource(req), getContext().dispatcher()).to(getSender());
+        // For scaling-down the decision requires getting idle hosts first.
+        // if enableSkuSpecUpgrade is true, first fetch the latest spec to override the sku spec during upgrade
```

Comment:
Instead of fetching this, could we send this as an argument to the workflow?

Reply:
it is sending the spec over the wire.